js/base/main.js

```diff
@@ -446,6 +460,15 @@ class RipeCommonsMainPlugin extends RipeCommonsPlugin {
                     this.$bus.trigger("format_change", format)
                 );
 
+                // registers a watch operation on all sdk options fields
+                // the data store so that the change is propagated to the ripe
+                // instance and then to the user interface
+                this.$store.watch(this.$store.getters.getSdkOptions, () => {
+                    const allOptions = this.$store.getters.getOptions();

```

Comment:
The argument of this callback is only these options:
<img width="322" alt="imagem" src="https://user-images.githubusercontent.com/24736423/81388533-53d20b00-9110-11ea-8884-841b3b33ae07.png">

We want the full state object
<img width="338" alt="imagem" src="https://user-images.githubusercontent.com/24736423/81388584-63e9ea80-9110-11ea-900b-7a52dbeab340.png">

Otherwise we would need to create a `setSomeOptions` method in `ripe-sdk` that would receive only the options we would want to change and leave the others. Currently the `setOptions` will replace the entire this.options object instead of merging the new options into the current ones.

We thought this solution would be less intrusive as the latter would require a change in `ripe-sdk` to create this new merge `setSomeOptions` method.

Reply:
Why don't you just `sdk.setOptions({ ...sdk.options, ...yourNewOptions })`?